### ISSUE TITLE
Loading a model not working correctly giving incorrect loss values

### ISSUE THREAD

**Issue by NIKHILDUGAR** [msed-Ebrahimi/ARoFace#8]
I trained a model  for 15 epochs and then my machine shut down, I want to continue this training but the loss shown is same as that of 0th epoch. the checkpoint was saved as:-
`                checkpoint = {
                    "epoch": epoch,
                    "global_step": global_step,
                    "state_dict_backbone": backbone.module.state_dict(),
                    "state_dict_softmax_fc": module_partial_fc.state_dict(),
                    "state_optimizer": opt.state_dict(),
                    "state_lr_scheduler": lr_scheduler.state_dict()
                }
                torch.save(checkpoint, os.path.join(cfg.output, f"checkpoint_global_step}___{loss}.pt"))
`
and loaded as :-
`    if cfg.resume:
        dict_checkpoint = torch.load("/path/to/model.pt")
        start_epoch = dict_checkpoint["epoch"]
        global_step = dict_checkpoint["global_step"]
        backbone.module.load_state_dict(dict_checkpoint["state_dict_backbone"])
        module_partial_fc.load_state_dict(dict_checkpoint["state_dict_softmax_fc"])
        opt.load_state_dict(dict_checkpoint["state_optimizer"])
        lr_scheduler.load_state_dict(dict_checkpoint["state_lr_scheduler"])`

And when the loss gets printed i am getting the same loss as of 0th Epoch.
Would love to know if I am missing something when loading model or saving the model.

**Comment by msed-Ebrahimi** [comment]
Hi, please double-check the saved checkpoints to see if it is saved as it should be (all states should be there including backbone, softmax weights, etc). Normally, save_all_states=false is in the config file, and only the backbone weights are saved in every epoch.

**Comment by NIKHILDUGAR** [comment]
Thanks for the quick response. I do believe the  save_all_state was true and if i run 
`a,b = backbone.module.load_state_dict(dict_checkpoint["state_dict_backbone"],strict=  True)
print("a============",a)
print("b============",b)`

 a,b are empty lists  so the models should be loaded properly.

**Comment by NIKHILDUGAR** [comment]
can you mention all the keys necessary for it to be loaded in a way to actually resume correctly?
